content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md

```diff
@@ -662,53 +662,99 @@ updates:
+| Option | Supported by | Action |
+|--------|--------------|--------|
+| `auto` | Try to differentiate between between apps and libraries. Use `increase` for apps and `widen` for libraries.|
+| `increase`| Always increase the minimum version requirement to match the new version. If a range already exists, typically this only increases the lower bound. |
+| `increase-if-necessary` | Leave the constraint if the original constraint allows the new version, otherwise, bump the constraint. |
+| `lockfile-only` | Only create pull requests to update lockfiles. Ignore any new versions that would require package manifest changes. |
+| `widen`| Widen the allowed version requirements to include both the new and old versions, when possible. Typically this only increases the maximum allowed version requirement. |
+
+Example:
+
+<table>
+    <thead>
+        <tr>
+            <th align="center">Constraint</th>
+            <th align="center">Current Version</th>
+            <th align="center">New Version</th>
+            <th>Strategy</th>
+            <th>New Constraint</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan="3" align="center"><code>^1.0.0</code></td>
+            <td rowspan="3" align="center"><code>1.0.0</code></td>
+            <td rowspan="3" align="center"><code>1.2.0</code></td>
+            <td align="center"><code>'widen'</code></td>
+            <td><code>^1.0.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase'</code></td>
+            <td><code>^1.2.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase-if-necessary'</code></td>
+            <td><code>^1.0.0</code></td>
+        </tr>
+      <tr>
+            <td rowspan="3" align="center"><code>^1.0.0</code></td>
+            <td rowspan="3" align="center"><code>1.0.0</code></td>
+            <td rowspan="3" align="center"><code>2.0.0</code></td>
+            <td align="center"><code>'widen'</code></td>
+            <td><code>&gt;=1.0.0 &lt;3.0.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase'</code></td>
+            <td><code>^2.0.0</code></td>
+        </tr>
+        <tr>
+            <td align="center"><code>'increase-if-necessary'</code></td>
+            <td><code>^2.0.0</code></td>
+        </tr>
+    </tbody>
+</table>
 
 Use the `versioning-strategy` option to change this behavior for supported package managers.
 
 {% data reusables.dependabot.option-affects-security-updates %}
 
 Available update strategies
 
-| Option | Supported by | Action |
-|--------|--------------|--------|
-| `lockfile-only` | `bundler`, `cargo`, `composer`, `mix`, `npm`, `pip` | Only create pull requests to update lockfiles. Ignore any new versions that would require package manifest changes. |
-| `auto` | `bundler`, `cargo`, `composer`, `mix`, `npm`, `pip` | Follow the default strategy described above.|
```

Comment:
I was a little confused by the old docs... they imply all these ecosystems support `auto` which implies these ecosystems also support `widen` / `increase`... but not all these ecosystems are listed under `widen` / `increase`... is that an omission or am I misunderstanding how `auto` works?

Reply:
The code was difficult to trace, so I finally restored to testing this by setting up a test repo and trying different options. Thankfully the `dependabot.yml` configuration validator CI check would not only flag wrong values, but also list the allowed values, so it wasn't quite as painful as it could have been.